lib/fluent/config/configure_proxy.rb

```diff
@@ -14,8 +14,6 @@
 #    limitations under the License.
 #
 
-require 'fluent/configurable'
```

Comment:
`ConfigureProxy` need to lookup types from registry.
Could you add a constructor argument of lambda to lookup types, and feed it to `ConfigureProxy.new` by `Configure::ClassMethods#configure_proxy`.
```
# configure.rb
type_lookup = ->(type){ Fluent::Configurable.lookup_type(type) }
proxy = Fluent::Config::ConfigureProxy.new(mod_name, type_lookup, required: true, multi: false)

# configure_proxy.rb
type = :string if type.nil?
block ||= @type_lookup.call(type)
```